batch2/create-batch-tables.sql

```diff
@@ -120,6 +122,14 @@ CREATE INDEX batch_attributes_key_value ON `batch_attributes` (`key`, `value`(25
 
 DELIMITER $$
 
+CREATE TRIGGER instances_before_update BEFORE UPDATE on instances
+FOR EACH ROW
+BEGIN
+  IF OLD.time_deactivated IS NOT NULL AND (NEW.time_deactivated IS NULL OR NEW.time_deactivated > OLD.time_deactivated) THEN
+    SET NEW.time_deactivated = OLD.time_deactivated;
```

Comment:
This is good enough for now, but what I really want to calculate is the cost of the instance, and the deactivation time is slightly conservative.  That said, I spent a little more time searching through the instance activity logs and it doesn't seem like there is a way to figure out when an instance stops (e.g. if it shut itself down or was preempted), although we do get notified when the deletion finishes.  Recording deletion times would be just as bad in the other direction.  Hmm.